Compilation errors produce stacktraces

<!--
If this is a security issue, please report it following the
[security reporting procedure](https://github.com/model-checking/kani/security/policy).
-->

<!--
Thank you for filing a bug report! 🐛
Please provide a short summary of the issue, along with the information necessary to replicate.
-->

When `RUST_BACKTRACE=1` is set, Kani produces a stacktrace when there is a compile error. This is annoying for packages that have `RUST_BACKTRACE=1` turned on by default.

Steps to reproduce:

1. `cargo new foo`
2. Put the following in `src/main.rs`:
```rust
fn f(x: i32) {
    assert!(x >= 0);
}

#[cfg_attr(kani, kani::proof)]
fn main() {
    f('c');
}
```
3. Set the `RUST_BACKTRACE` environment variable to 1 and run `cargo kani`, e.g. `RUST_BACKTRACE=1 cargo kani`. This produces:
```bash
$ RUST_BACKTRACE=1 cargo kani
   Compiling foo v0.1.0 (/home/ubuntu/examples/foo)
error[E0308]: mismatched types
 --> src/main.rs:7:7
  |
7 |     f('c');
  |     - ^^^ expected `i32`, found `char`
  |     |
  |     arguments to this function are incorrect
  |
note: function defined here
 --> src/main.rs:1:4
  |
1 | fn f(x: i32) {
  |    ^ ------
help: you can cast a `char` to an `i32`, since a `char` always occupies 4 bytes
  |
7 |     f('c' as i32);
  |           ++++++

For more information about this error, try `rustc --explain E0308`.
Error: "Failed to compile crate."
error: could not compile `foo` due to previous error
Error: cargo exited with status exit status: 101

Stack backtrace:
   0: anyhow::error::<impl anyhow::Error>::msg
             at /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.68/src/error.rs:83:36
   1: kani_driver::session::KaniSession::run_terminal
             at /home/ubuntu/git/kani/kani-driver/src/session.rs:141:13
   2: kani_driver::call_cargo::<impl kani_driver::session::KaniSession>::cargo_build
             at /home/ubuntu/git/kani/kani-driver/src/call_cargo.rs:124:17
   3: kani_driver::project::cargo_project
             at /home/ubuntu/git/kani/kani-driver/src/project.rs:152:19
   4: kani_driver::cargokani_main
             at /home/ubuntu/git/kani/kani-driver/src/main.rs:61:19
   5: kani_driver::main
             at /home/ubuntu/git/kani/kani-driver/src/main.rs:44:44
   6: core::ops::function::FnOnce::call_once
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/core/src/ops/function.rs:507:5
   7: std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/sys_common/backtrace.rs:121:18
   8: std::rt::lang_start::{{closure}}
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/rt.rs:166:18
   9: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/core/src/ops/function.rs:606:13
  10: std::panicking::try::do_call
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/panicking.rs:483:40
  11: std::panicking::try
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/panicking.rs:447:19
  12: std::panic::catch_unwind
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/panic.rs:137:14
  13: std::rt::lang_start_internal::{{closure}}
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/rt.rs:148:48
  14: std::panicking::try::do_call
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/panicking.rs:483:40
  15: std::panicking::try
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/panicking.rs:447:19
  16: std::panic::catch_unwind
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/panic.rs:137:14
  17: std::rt::lang_start_internal
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/rt.rs:148:20
  18: std::rt::lang_start
             at /rustc/c6fcdb690609769a240fc8ab0de0ce68d5ea7dba/library/std/src/rt.rs:165:17
  19: main
  20: __libc_start_main
  21: _start
$
```

This does not happen with `cargo build` or `cargo run`:
```bash
$ RUST_BACKTRACE=1 cargo run
   Compiling foo v0.1.0 (/home/ubuntu/examples/foo)
error[E0308]: mismatched types
 --> src/main.rs:7:7
  |
7 |     f('c');
  |     - ^^^ expected `i32`, found `char`
  |     |
  |     arguments to this function are incorrect
  |
note: function defined here
 --> src/main.rs:1:4
  |
1 | fn f(x: i32) {
  |    ^ ------
help: you can cast a `char` to an `i32`, since a `char` always occupies 4 bytes
  |
7 |     f('c' as i32);
  |           ++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `foo` due to previous error
$
```


with Kani version: 48569a261b8

--- a/kani-driver/src/main.rs
+++ b/kani-driver/src/main.rs
@@ -48,7 +48,7 @@ fn main() -> ExitCode {
 
     if let Err(error) = result {
         // We are using the debug format for now to print the all the context.
-        // We should consider create a standard for error reporting.
+        // We should consider creating a standard for error reporting.
         debug!(?error, "main_failure");
         util::error(&format!("{error:#}"));
         ExitCode::FAILURE

--- a/kani-driver/src/main.rs
+++ b/kani-driver/src/main.rs
@@ -4,6 +4,7 @@
 #![feature(array_methods)]
 
 use std::ffi::OsString;
+use std::process::ExitCode;
 
 use anyhow::Result;
 
@@ -39,10 +40,20 @@ mod unsound_experiments;
 /// The main function for the `kani-driver`.
 /// The driver can be invoked via `cargo kani` and `kani` commands, which determines what kind of
 /// project should be verified.
-fn main() -> Result<()> {
-    match determine_invocation_type(Vec::from_iter(std::env::args_os())) {
+fn main() -> ExitCode {
+    let result = match determine_invocation_type(Vec::from_iter(std::env::args_os())) {
         InvocationType::CargoKani(args) => cargokani_main(args),
         InvocationType::Standalone => standalone_main(),
+    };
+
+    if let Err(error) = result {
+        // We are using the debug format for now to print the all the context.
+        // We should consider create a standard for error reporting.
+        debug!(?error, "main_failure");
+        util::error(&format!("{error:#}"));
+        ExitCode::FAILURE
+    } else {
+        ExitCode::SUCCESS
     }
 }
 

--- a/kani-driver/src/util.rs
+++ b/kani-driver/src/util.rs
@@ -95,6 +95,13 @@ pub fn warning(msg: &str) {
     println!("{warning} {msg_fmt}")
 }
 
+/// Print a warning message. This will add a "warning:" tag before the message and style accordinly.
+pub fn error(msg: &str) {
+    let error = console::style("error:").bold().red();
+    let msg_fmt = console::style(msg).bold();
+    println!("{error} {msg_fmt}")
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;

--- a/kani-driver/src/util.rs
+++ b/kani-driver/src/util.rs
@@ -88,14 +88,14 @@ pub fn specialized_harness_name(linked_obj: &Path, harness_filename: &str) -> Pa
     alter_extension(linked_obj, &format!("for-{harness_filename}.out"))
 }
 
-/// Print a warning message. This will add a "warning:" tag before the message and style accordinly.
+/// Print a warning message. This will add a "warning:" tag before the message and style accordingly.
 pub fn warning(msg: &str) {
     let warning = console::style("warning:").bold().yellow();
     let msg_fmt = console::style(msg).bold();
     println!("{warning} {msg_fmt}")
 }
 
-/// Print a warning message. This will add a "warning:" tag before the message and style accordinly.
+/// Print an error message. This will add an "error:" tag before the message and style accordingly.
 pub fn error(msg: &str) {
     let error = console::style("error:").bold().red();
     let msg_fmt = console::style(msg).bold();

--- a/tests/cargo-kani/asm/global_error/doesnt_call_crate_with_global_asm.expected
+++ b/tests/cargo-kani/asm/global_error/doesnt_call_crate_with_global_asm.expected
@@ -1,3 +1,3 @@
 error: Crate crate_with_global_asm contains global ASM, which is not supported by Kani. Rerun with `--enable-unstable --ignore-global-asm` to suppress this error (**Verification results may be impacted**).
 error: could not compile `crate_with_global_asm` due to previous error
-Error: cargo exited with status exit status: 101
+error: cargo exited with status exit status: 101

--- a/tests/cargo-kani/simple-proof-annotation/main.expected
+++ b/tests/cargo-kani/simple-proof-annotation/main.expected
@@ -1,1 +1,1 @@
-Error: A proof harness named main was not found
+error: A proof harness named main was not found

--- a/tests/cargo-ui/unsupported-lib-types/proc-macro/expected
+++ b/tests/cargo-ui/unsupported-lib-types/proc-macro/expected
@@ -1,2 +1,2 @@
 Skipped the following unsupported targets: 'lib'.
-Error: No supported targets were found.
+error: No supported targets were found.

--- a/tests/expected/function-stubbing-no-harness/expected
+++ b/tests/expected/function-stubbing-no-harness/expected
@@ -1,1 +1,1 @@
-Error: A proof harness named foo was not found
+error: A proof harness named foo was not found

